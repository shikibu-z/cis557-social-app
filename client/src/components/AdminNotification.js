/* eslint-disable react/jsx-no-bind */
/* eslint-disable no-else-return */
import React from 'react';
import '../stylesheets/AdminNotifications.css';
import {
  Button,
} from 'react-bootstrap';

const lib = require('../api/fetchers');

function AdminNotification(props) {
  const { note, group } = props;

  function handleJoinRequest(e) {
    const action = e.currentTarget.textContent;
    if (action === 'Deny') {
      const data = { groupId: group.group.idGroups, userId: note.idUser_Action, approve: 0 };
      lib.resolveJoinRequest(data, note.idAdminNotifications);
      document.getElementById('approve').style.display = 'none';
      document.getElementById('deny').style.display = 'none';
    } else if (action === 'Approve') {
      const data = { groupId: group.group.idGroups, userId: note.idUser_Action, approve: 1 };
      lib.resolveJoinRequest(data, note.idAdminNotifications);
      document.getElementById('approve').style.display = 'none';
      document.getElementById('deny').style.display = 'none';
    }
  }

  function handleFlagRequest() {

  }

  function handleView() {
    // update read to 1
    return 1;
  }

  if (note.type === 'leave') {
    return (
      <div className="adminNotification">
        {note.action}
        {note.read_status === 0 && <Button onClick={handleView}>Viewed</Button>}
      </div>
    );
  } else if (note.type === 'join') {
    return (
      <div className="adminNotification">
        {note.action}
        {note.read_status === 0 && <Button id="approve" onClick={handleJoinRequest}>Approve</Button>}
        {note.read_status === 0 && <Button id="deny" onClick={handleJoinRequest}>Deny</Button>}
      </div>
    );
  }
  return (
    <div className="adminNotification">
      {note.action}
      {note.read_status === 0 && <Button onClick={handleFlagRequest}>View</Button>}
    </div>
  );
}

export default AdminNotification;
