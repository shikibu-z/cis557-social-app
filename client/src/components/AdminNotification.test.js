/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import AdminNotification from './AdminNotification';

test('Chat matches snapshot', () => {
  const component = renderer.create(<AdminNotification group="" note="" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
