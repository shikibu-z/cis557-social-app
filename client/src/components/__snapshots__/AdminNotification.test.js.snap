// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat matches snapshot 1`] = `
<div
  className="adminNotification"
/>
`;
